lib/initializer.rb

```diff
@@ -0,0 +1,47 @@
+require_relative './game.rb'
+require_relative './board/board.rb'
+require_relative './board/board_checker.rb'
+require_relative './players/ai.rb'
+require_relative './players/human.rb'
+require_relative './players/player.rb'
+require_relative './ui/user_interface.rb'
+require_relative './ui/script.rb'
+
+def initializer(size, empty_char)
+  board = Board.new(size, empty_char)
+  checker = BoardChecker.new(size, empty_char)
+  script = Script.new
+  user_interface = UserInterface.new(board, script)
+
+  user_interface.welcome
+  mode_selection = user_interface.what_mode?
+
+  unless(mode_selection == 'computerVScomputer')
+    player_selection = user_interface.who_goes_first?
+  end
+
```

Comment:
Though I did mention that setting up all the dependencies could be completed procedurally, I think that some of the "procedure" could be wrapped up into their own method. Something that just returns/yields some game options.